clar2wasm/tests/wasm-generation/hashing.rs

```diff
@@ -0,0 +1,32 @@
+use clar2wasm::tools::crosscheck_compare_only;
+use clarity::vm::types::{SequenceSubtype, TypeSignature};
+use proptest::strategy::{Just, Strategy};
+use proptest::{prop_oneof, proptest};
+
+use crate::PropValue;
+
+const HASH_FUNC: [&str; 5] = ["hash160", "keccak256", "sha256", "sha512", "sha512/256"];
+
+fn strategies_for_hashing() -> impl Strategy<Value = TypeSignature> {
+    prop_oneof![
+        Just(TypeSignature::IntType),
+        Just(TypeSignature::UIntType),
+        (0u32..=300).prop_map(|s| TypeSignature::SequenceType(SequenceSubtype::BufferType(
+            s.try_into().unwrap()
+        ))),
+    ]
+}
+
+proptest! {
+    #![proptest_config(super::runtime_config())]
+
+    #[test]
+    fn crossprop_hashing(val in strategies_for_hashing().prop_flat_map(PropValue::from_type))
+    {
+        for func in &HASH_FUNC {
+            crosscheck_compare_only(
+                &format!("({func} {val})")
+            )
+        }
```

Comment:
If the test fail, proptest should give us the the generated value, but it won't tell which hash function actually failed. I would split this loop into different tests for each function.

Reply:
Good point, separated (I took inspiration from bitwise/arithmetic and some other for the aggregated version). Also replaced `crosscheck_compare_only` with `crosscheck_validate` to ensure we expect results and not compilation errors, WDYT?

Reply:
I had no idea there was a `crosscheck_validate` function, that looks like a good idea :D